src/Entity/OgRole.php

```diff
@@ -165,4 +200,123 @@ public function save() {
+    // tampered with.
+    $is_locked_property = in_array($property_name, [
+      'id',
+      'role_type',
+      'group_type',
+      'group_bundle',
+    ]);
+    $is_default_role = $this->getRoleType() !== self::ROLE_TYPE_STANDARD;
+    if ($is_locked_property && $is_default_role && !$this->isNew()) {
+      throw new OgRoleRequiredException("The $property_name of the default roles 'non-member' and 'member' cannot be changed.");
+    }
+    return parent::set($property_name, $value);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function delete() {
+    // The default roles are required. Prevent them from being deleted for as
+    // long as the group still exists.
+    if (in_array($this->id(), [self::ANONYMOUS, self::AUTHENTICATED]) && $this->groupManager()->isGroup($this->getGroupType(), $this->getGroupBundle())) {
+      throw new OgRoleRequiredException('The default roles "non-member" and "member" cannot be deleted.');
+    }
+    parent::delete();
+  }
+
+  /**
+   * Returns default properties for each of the standard role names.
+   *
+   * @param string $role_name
+   *   A role name, one of OgRoleInterface::ANONYMOUS,
+   *   OgRoleInterface::AUTHENTICATED, OgRoleInterface::ADMINISTRATOR.
+   *
+   * @return array
+   *   An array of default properties, to pass to OgRole::create().
+   */
+  public static function getDefaultProperties($role_name) {
+    if (!in_array($role_name, [
+      self::ANONYMOUS,
+      self::AUTHENTICATED,
+      self::ADMINISTRATOR,
+    ])) {
+      throw new \InvalidArgumentException("$role_name is not a default role name.");
+    }
+
+    $default_properties = [
+      self::ANONYMOUS => [
+        'role_type' => OgRoleInterface::ROLE_TYPE_IMMUTABLE,
+        'label' => 'Non-member',
+        'permissions' => ['subscribe'],
+      ],
+      self::AUTHENTICATED => [
+        'role_type' => OgRoleInterface::ROLE_TYPE_IMMUTABLE,
+        'label' => 'Member',
+        'permissions' => ['unsubscribe'],
+      ],
+      self::ADMINISTRATOR => [
+        'role_type' => OgRoleInterface::ROLE_TYPE_STANDARD,
+        'label' => 'Administrator',
+        'permissions' => [
+          'add user',
+          'administer group',
+          'approve and deny subscription',
+          'manage members',
+          'manage permissions',
+          'manage roles',
+          'update group',
+        ],
+      ],
+    ];
+
+    return $default_properties[$role_name];
+  }
+
+  /**
+   * Maps role names to role types.
+   *
+   * The 'anonymous' and 'authenticated' roles should not be changed or deleted.
+   * All others are standard roles.
+   *
+   * @param string $role_name
+   *   The role name for which to return the type.
+   *
+   * @return string
+   *   The role type, either OgRoleInterface::ROLE_TYPE_IMMUTABLE or
+   *   OgRoleInterface::ROLE_TYPE_STANDARD.
+   */
+  public static function getRoleTypeByName($role_name) {
+    switch ($role_name) {
```

Comment:
Minor, but maybe a little simpler? up to you :)

`return in_array($role_name, [OgRoleInterface::ANONYMOUS, OgRoleInterface::AUTHENTICATED]) ? OgRoleInterface::ROLE_TYPE_IMMUTABLE : OgRoleInterface::ROLE_TYPE_STANDARD;`